integration-cli/docker_cli_build_test.go

```diff
@@ -7327,3 +7312,22 @@ func (s *DockerSuite) TestBuildWindowsEnvCaseInsensitive(c *check.C) {
 		c.Fatalf("Case insensitive environment variables on Windows failed. Got %s", res)
 	}
 }
+
+// Test case for #29005
+func (s *DockerSuite) TestBuildRunCommentLineAfterEscape(c *check.C) {
+	name := "testbuildcommentlineafterescape"
+
+	// Note comment is defined as starting `#` at the first pos fo the line
+	_, out, err := buildImageWithOut(name,
+		`
+FROM busybox
+RUN \
+# my comment
+# some other comment
+# and some script, etc
+echo bar
```

Comment:
Thanks @thaJeztah. The PR has been updated using the test case above. Please take a look and let me know if there are any issues.